lib/effectiveCallNumber.js

```diff
@@ -14,23 +18,42 @@ import { get } from 'lodash';
  *   <EffectiveCopy>
  *
  * @param {object} item an item record as returned from /inventory/items/${item-id}
+ * @param {object} item returned from circulation/loans with structure {item: {}}
  * @return {string} the effective call number
  */
 export default function effectiveCallNumber(item) {
-  const prefix = get(item, 'effectiveCallNumberComponents.prefix') ||
-    get(item, 'callNumberComponents.prefix', '');
-  const suffix = get(item, 'effectiveCallNumberComponents.suffix') ||
-    get(item, 'callNumberComponents.suffix', '');
-  const callNumber = get(item, 'effectiveCallNumberComponents.callNumber') ||
-    get(item, 'callNumberComponents.callNumber', '');
+  if (has(item, 'item')) {
```

Comment:
@annamelnyk if we assume that the order of the elements is the same in both cases (when `item.item` is present and when it's  not) could we do a simple test at the beginning similar to:

````js
const rootItem = item?.item ?? item;
````

that way you wouldn't need to repeat the same code in both cases and just do:
````js
const result = [
  prefix,
  callNumber,
  suffix,
  get(item, 'volume', ''),
  get(item, 'enumeration', ''),
  get(item, 'chronology', ''),
]
````
and  you could just add copyNumbers at the end:
```js
if (!has(item, 'item')) {
  result.push(get(item, 'copyNumbers[0]', ''))
}
````
Assuming we need copyNumbers in cases when `item.item` is not present.

I would assume we want the same format in both cases when `item.item` is present and when it's not.

Also we now have access to optional chaining:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

This could be a good opportunity to get rid of `get` from lodash. What do you think?

Reply:
Thanks for review @mkuklis, currently I am waiting for final deсision about fields order from PO, and then I will make changes. It would be great to use optional chaining operator here.